src/LA/system_solvers_small.cpp

```diff
@@ -353,6 +363,62 @@ backwardSubstitution(U, z, x);
 
 return; };
 
+// -------------------------------------------------------------------------- //
+/*!
+    Solve a linear system using Lapacke_dgesv (partial pivoting
+    and LU factorization). Containers MUST be correctly sized.
+    This routine works on the entire matrix, the number of considered equations
+    is equal to the leading dimension of the matrix, the same is for the r.h.s.
+    Multiple r.h.s. are not allowed. Permutation matrix not provided.
+    This function is intended for solution computation ONLY.
+
+    \param[in] layout matrix layout (possible values are ROW_MAJOR and COL_MAJOR)
+    \param[in,out] A in input the coeffs of square matrix (linear), in output the permuted factors L and U.
+    \param[in,out] B in input r.h.s. of the linear system, in output the solution.
+    \return information on the execution of LAPACKE_dgesv, see Lapack documentation
+*/
+int solveLU(
+    int                              layout,
+    std::vector<double>                  &A,
+    std::vector<double>                  &B
+) {
+
+    std::vector<int> ipiv(B.size());
+    int info = solveLU(layout, B.size(), A.data(), B.data(),ipiv.data());
+    return info;
+
+};
+
+// -------------------------------------------------------------------------- //
+/*!
+    Solve a linear system using Lapacke_degesv (partial pivoting
+    and LU factorization).
+    This routine works on the entire matrix, the number of considered equations
+    is equal to the leading dimension of the matrix, the same is for the r.h.s.
+    Multiple r.h.s. are not allowed.
+    This function is intended for both solution and factorization computation.
+
+    \param[in] layout matrix layout (possible values are ROW_MAJOR and COL_MAJOR).
+    \param[in] matrixOrder number of equations in linear system.
+    \param[in,out] A in input the coeffs of square matrix (linear), in output the the factors L and U, A = P * L * U.
+    \param[in,out] B in input r.h.s. of the linear system, in output the solution.
+    \param[out] ipiv the pivot indices that define the permutation matrix P, see [1].
```

Comment:
I don't see where reference [1] is defined.

Reply:
I forgot to correct, referring Lapack Doc.